core/client/fs/src/main/java/alluxio/client/file/cache/CacheManagerOptions.java

```diff
@@ -0,0 +1,214 @@
+        .setMaxEvictionRetries(conf.getInt(PropertyKey.USER_CLIENT_CACHE_EVICTION_RETRIES))
+        .setPageSize(conf.getBytes(PropertyKey.USER_CLIENT_CACHE_PAGE_SIZE))
+        .setQuotaEnabled(conf.getBoolean(PropertyKey.USER_CLIENT_CACHE_QUOTA_ENABLED))
+        .setCacheEvictorOptions(cacheEvictorOptions)
+        .setPageStoreOptions(PageStoreOptions.create(conf));
+    return options;
+  }
+
+  /**
+   * @param conf
+   * @return instance of CacheManagerOptions
+   */
+  public static CacheManagerOptions createForWorker(AlluxioConfiguration conf) {
+    CacheEvictorOptions cacheEvictorOptions = new CacheEvictorOptions()
+        .setEvictorClass(conf.getClass(PropertyKey.WORKER_PAGE_STORE_EVICTOR_CLASS))
+        .setIsNondeterministic(
+            conf.getBoolean(PropertyKey.WORKER_PAGE_STORE_EVICTOR_NONDETERMINISTIC_ENABLED));
+    CacheManagerOptions options = new CacheManagerOptions()
+        .setAsyncRestoreEnabled(
+            conf.getBoolean(PropertyKey.WORKER_PAGE_STORE_ASYNC_RESTORE_ENABLED))
+        .setAsyncWriteThreads(conf.getInt(PropertyKey.WORKER_PAGE_STORE_ASYNC_WRITE_THREADS))
+        .setIsAsyncWriteEnabled(
+            conf.getBoolean(PropertyKey.WORKER_PAGE_STORE_ASYNC_WRITE_ENABLED))
+        .setMaxEvictionRetries(conf.getInt(PropertyKey.WORKER_PAGE_STORE_EVICTION_RETRIES))
+        .setPageSize(conf.getBytes(PropertyKey.WORKER_PAGE_STORE_PAGE_SIZE))
+        .setQuotaEnabled(conf.getBoolean(PropertyKey.WORKER_PAGE_STORE_QUOTA_ENABLED))
+        .setCacheEvictorOptions(cacheEvictorOptions)
+        .setPageStoreOptions(PageStoreOptions.createForWorkerPageStore(conf));
+    return options;
+  }
+
+  /**
+   * Constructor.
+   */
+  public CacheManagerOptions() {
+  }
+
+  /**
+   * @return if async restore is enabled
+   */
+  public boolean isAsyncRestoreEnabled() {
+    return mAsyncRestoreEnabled;
+  }
+
+  /**
+   * @return if async write is enabled
+   */
+  public boolean isAsyncWriteEnabled() {
+    return mAsyncWriteEnabled;
+  }
+
+  /**
+   * @return number of thread for async write
+   */
+  public int getAsyncWriteThreads() {
+    return mAsyncWriteThreads;
+  }
+
+  /**
+   * @return if quota is enabled
+   */
+  public boolean isQuotaEnabled() {
+    return mQuotaEnabled;
+  }
+
+  /**
+   * @return max eviction retires
+   */
+  public int getMaxEvictionRetries() {
+    return mMaxEvictionRetries;
+  }
+
+  /**
+   * @return the page size
+   */
+  public long getPageSize() {
+    return mPageSize;
+  }
+
+  /**
+   * @return the list of PageStoreOptions
+   */
+  public List<PageStoreOptions> getPageStoreOptions() {
+    return mPageStoreOptions;
+  }
+
+  /**
+   * @return the options of cache evictor
+   */
+  public CacheEvictorOptions getCacheEvictorOptions() {
+    return mCacheEvictorOptions;
+  }
+
+  /**
+   * @param isAsyncRestoreEnabled
+   * @return the updated options
+   */
+  public CacheManagerOptions setAsyncRestoreEnabled(boolean isAsyncRestoreEnabled) {
```

Comment:
Is CacheManagerOptions will not be modified once created and is unmodifiable?
Alluxio's convention of using set methods and making all options modifiable may be better to change.

Reply:
we could make options to be immutable, but as you said,  looks like alluxio convention is making the options modifiable.  which way do you prefer?

Reply:
Rongrong discussed with us before that she wants to create a new convention to make all options immutable. The newly added one is better to be immutable. This option class has the advantages that it does not need to be serialized or deserialized